tools/docker/scripts/test_python.sh

```diff
@@ -1,62 +1,54 @@
-#!/bin/bash
+#!/bin/bash -e
 
 # author: Ole Schuett
 
-set -e
-
 run_selftests() {
     PYTHON=$1
-    shift
-    BASEDIR=`pwd`
-    VERSION=`$PYTHON -V 2>&1 | tr -d '\n'`
+    SCRIPTS=$2
+    BASEDIR=$(pwd)
+    VERSION=$(${PYTHON} -V 2>&1 | tr -d '\n')
     echo ""
     echo "========== Running ${VERSION} Tests =========="
     #find ./src/ ./tools/ -name "*.pyc" -exec rm {} \;
-    for i in $@ ; do
+    for i in $SCRIPTS ; do
         set +e #disable error trapping
-        if [[ `head -n1 $i` =~ "python3" && "$PYTHON" != "python3" ]]; then
+        if [[ $(head -n1 "$i") =~ "python3" && "$PYTHON" != "python3" ]]; then
           echo "Skipping $i - it's Python3 only."
           continue
         fi
-        echo "Selftesting" $i
-        cd $(dirname $i)
-        $PYTHON -B ./$(basename $i) --selftest
-        if [ $? -ne 0 ]; then
+        echo "Selftesting $i"
+        cd "$(dirname "$i")"
+        SCRIPT=$(basename "$i")
+        if ! $PYTHON -B "./${SCRIPT}" --selftest ; then
             echo "Selftest of $i failed"
             ERRORS=$((ERRORS+1))
         fi
         set -e #re-enable error trapping
-        cd $BASEDIR
+        cd "$BASEDIR"
     done
 }
 
-
 #===============================================================================
-echo -e "\n========== Copying Changed Files =========="
-mkdir /opt/cp2k-local-rw/
-cp -r /opt/cp2k-local/tools /opt/cp2k-local-rw/tools
-cp -r /opt/cp2k-local/src /opt/cp2k-local-rw/src
-cd /opt/cp2k-local-rw/
-
-
 ERRORS=0
 
+cd /workspace/cp2k
+
 # find executable python scripts
-ALL_SCRIPTS=`find ./src/ ./tools/ -name "*.py"  -executable`
-ESSENTIAL_SCRIPTS=`find ./tools/build_utils -name "*.py"  -executable`
+ALL_SCRIPTS=$(find ./src/ ./tools/ -name "*.py"  -executable)
+ESSENTIAL_SCRIPTS=$(find ./tools/build_utils -name "*.py"  -executable)
 ESSENTIAL_SCRIPTS="$ESSENTIAL_SCRIPTS ./src/dbcsr/libsmm_acc/libcusmm/generate.py"
 
 # python 2.6
-run_selftests python2.6 $ESSENTIAL_SCRIPTS
+run_selftests python2.6 "${ESSENTIAL_SCRIPTS}"
```

Comment:
Please drop 2.6 testing: Python 2.6 is EOL and popular projects have even started dropping Python 2.7 support

Reply:
It probably comes at no surprise to you that I'm reluctant to drop Python 2.6 support ;-)

However, we should really find a better way to settle these kind of arguments. So, I'll propose we simply ask our users (#90).